api/src/main/java/ai/djl/util/NativeResource.java

```diff
@@ -14,58 +14,42 @@
 
 import com.sun.jna.Pointer;
 
-import java.util.concurrent.atomic.AtomicReference;
-
 /**
- * {@code NativeResource} is an internal class for {@link AutoCloseable} blocks of memory created in
- * the different engines.
+ * {@code NativeResource} is an interface for {@link AutoCloseable} blocks of memory created in the
+ * different engines.
  *
  * @param <T> the resource that could map to a native pointer or java object
  */
-public abstract class NativeResource<T> implements AutoCloseable {
-
-    protected final AtomicReference<T> handle;
-    private String uid;
-
-    protected NativeResource(T handle) {
-        this.handle = new AtomicReference<>(handle);
-        uid = handle.toString();
-    }
-
+public interface NativeResource<T> extends AutoCloseable {
```

Comment:
Does this need to be interface?

Reply:
I think yes. By fassading PtNDArray by a dynamic proxy the whole interface of PtNDArray must be a Java interface. NativeResource<T> is part of it.